### PR TITLE
Fix Iceberg GetPartitions Sort order for DateCreated timestamp

### PR DESCRIPTION
**Pull request by raveeram**

### CODE REVIEW

File: metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/sql/HiveConnectorFastPartitionService.java [hunk 1]
```diff
@@ -65,6 +65,8 @@
  */
 @Slf4j
 public class HiveConnectorFastPartitionService extends HiveConnectorPartitionService {
+    private static final String SORT_BY_DATE_CREATED = "dateCreated";
+
```
Comment (zhljen) [inline metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/sql/HiveConnectorFastPartitionService.java]: Is this constant used anywhere else? If so, you can put into our constant class.

File: metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/sql/HiveConnectorFastPartitionService.java [hunk 2]
```diff
@@ -389,9 +391,17 @@ public void deletePartitions(
             .collect(Collectors.toList());
 
         if (sort != null) {
-            //it can only support sortBy partition Name
-            final Comparator<PartitionInfo> nameComparator = Comparator.comparing(p -> p.getName().toString());
-            ConnectorUtils.sort(filteredPartitionList, sort, nameComparator);
+            if (sort.hasSort() && sort.getSortBy().equalsIgnoreCase(SORT_BY_DATE_CREATED)) {
+                final Comparator<PartitionInfo> dateCreatedComparator = Comparator.comparing(
+                    p -> p.getAudit().getCreatedDate(),
+                    Comparator.nullsLast(Date::compareTo));
```
Comment (zhljen) [inline metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/sql/HiveConnectorFastPartitionService.java]: Can you please check the p.getAudit() will not have a null pointer?

File: metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/sql/HiveConnectorFastPartitionService.java [hunk 3]
```diff
@@ -389,9 +391,17 @@ public void deletePartitions(
...
+
+                ConnectorUtils.sort(filteredPartitionList, sort, dateCreatedComparator);
+            } else {
+                // Sort using the partition name by default
+                final Comparator<PartitionInfo> nameComparator = Comparator.comparing(p -> p.getName().toString());
+                ConnectorUtils.sort(filteredPartitionList, sort, nameComparator);
+            }
```
Comment (zhljen) [inline metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/sql/HiveConnectorFastPartitionService.java]: Good with the fallback.

File: metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/sql/HiveConnectorFastPartitionService.java (hunk 1 above)
Comment (raveeram) [reply metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/sql/HiveConnectorFastPartitionService.java]: Just in the tests. Looks like there's one defined in DirectSqlGetPartition already. Updated to use that.

File: metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/sql/HiveConnectorFastPartitionService.java (hunk 2 above)
Comment (raveeram) [reply metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/sql/HiveConnectorFastPartitionService.java]: Updated.

**Pull request merged**